### PR TITLE
Add WebIDLs for experimental AudioContext sinkId, setSinkId

### PR DESCRIPTION
**Pull request by tylerbarker**
Hey, I'm working on some audio stuff and found these features I need were missing - thought it might be useful to contribute? Let me know if I'm missing something or have done this incorrectly. Cheers ✌️

### CODE REVIEW

**Comment by tylerbarker** [comment]
Okay what I can gather from the failing CI is there's an issue because I mistakenly hard-coded the experimental feature decorators in `gen_AudioContext.rs`.

I can see now they need to be included in the `unstable/` directory instead, how would I go about that given AudioContext is already stable? Is there a way I can write a partial IDL in `unstable/` which I can include in the main IDL?

**Comment by daxpedda** [comment]
Yes, you should be able to use [`partial interface`](https://webidl.spec.whatwg.org/#dfn-partial-interface), it's used quite often in `wasm-bindgen` already.

Let me know if you encounter any problems.

File: crates/web-sys/webidls/unstable/AudioContext.webidl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+partial interface AudioContext {
+  readonly        attribute DOMString sinkId;
+
+  [Throws]
```
Comment (daxpedda) [inline crates/web-sys/webidls/unstable/AudioContext.webidl:4]: Where did you get that this throws?
The spec doesn't mention it: https://webaudio.github.io/web-audio-api/#dom-audiocontext-setsinkid-domstring-or-audiosinkoptions-sinkid.
Chromium doesn't do it either: https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/interfaces/webaudio.idl;l=83?q=setsinkid&ss=chromium%2Fchromium%2Fsrc.

File: crates/web-sys/webidls/unstable/AudioContext.webidl [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  Promise<undefined> setSinkId(DOMString sinkId);
```
Comment (daxpedda) [inline crates/web-sys/webidls/unstable/AudioContext.webidl:5]:
```suggestion
  [SecureContext] Promise<undefined> setSinkId ((DOMString or AudioSinkOptions) sinkId);
```
We should follow the specification.

File: crates/web-sys/webidls/unstable/AudioContext.webidl [hunk 3]
```diff
@@ -0,0 +1,6 @@
+partial interface AudioContext {
+  readonly        attribute DOMString sinkId;
```
Comment (daxpedda) [inline crates/web-sys/webidls/unstable/AudioContext.webidl:2]:
```suggestion
  [SecureContext] readonly attribute (DOMString or AudioSinkInfo) sinkId;
```

**Review by daxpedda — CHANGES_REQUESTED** [review]
We should implement the rest of the interface too:
- `AudioSinkInfo`
- `AudioSinkOptions`
- `AudioSinkType`
- `BaseAudioContext.onsinkchange`
- `AudioContextOptions.sinkId`

File: crates/web-sys/webidls/unstable/AudioContext.webidl (hunk 1 above)
Comment (tylerbarker) [reply crates/web-sys/webidls/unstable/AudioContext.webidl:4]: Ah, yeah I've not properly understood the mechanism then, I was under the impression promises have the possibility of throwing and I needed it here like I saw else - will remove 👍

Comment (daxpedda) [reply crates/web-sys/webidls/unstable/AudioContext.webidl:4]: Just to clarify, `Promise`s throwing or not is handled by the `Promise` type itself, no need to label it in the WebIDL at all.

Comment (tylerbarker) [reply crates/web-sys/webidls/unstable/AudioContext.webidl:4]: FWIW this was what I was looking at: [MDN setSinkId Exceptions](https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/setSinkId#exceptions)

Comment (tylerbarker) [reply crates/web-sys/webidls/unstable/AudioContext.webidl:4]: > Promises throwing or not is handled by the Promise type itself, no need to label it in the WebIDL at all.
That's helpful, thanks!

Comment (daxpedda) [reply crates/web-sys/webidls/unstable/AudioContext.webidl:4]: > FWIW this was what I was looking at: [MDN setSinkId Exceptions](https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/setSinkId#exceptions)

I just tried it myself, I can confirm that the exceptions listed here are rejected promises and aren't thrown on the function itself.

**Comment by daxpedda** [comment]
To generate the WebIDL correctly you have to run the following:
```
cd crates/web-sys
cargo run --release --package wasm-bindgen-webidl -- webidls src/features ./Cargo.toml
```

**Review by daxpedda — CHANGES_REQUESTED** [review]
LGTM!
Missing an entry in the changelog.

**Comment by daxpedda** [comment]
Thanks @tylerbarker!

I just rebased and fixed the changelog.

**Pull request merged**